packages/mobile/src/screens/home/components/product/product.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC } from 'react';
+import { useTranslation, useAppSelector, useCustomTheme } from '~/hooks/hooks';
+import { selectProductById } from '~/store/products/selectors';
+import { Button, ClockIcon, Image, Text, View } from '~/components/components';
+import { globalStyles } from '~/styles/styles';
+import { ProductType, ProductDto } from '@vse-bude/shared';
+import { formatPrice, getTimeToEvent } from '~/helpers/helpers';
+import { styles } from './styles';
+
+type Props = {
+  productId: string;
+};
+
+const Product: FC<Props> = ({ productId }) => {
+  const { imageLinks, title, description, price, type, endDate } =
+    useAppSelector((state) =>
+      selectProductById(state, productId),
+    ) as ProductDto;
+  const duration = getTimeToEvent(endDate);
+  const auction = ProductType.AUCTION;
+  const { colors } = useCustomTheme();
+  const { t } = useTranslation();
+
+  return (
+    <View style={[styles.container, globalStyles.boxShadow]}>
+      <View style={[styles.imgWrapper, { backgroundColor: colors.card }]}>
+        <Image source={{ uri: imageLinks[0] }} style={styles.img} />
+
+        {type === auction && (
+          <View
+            style={[
+              styles.time,
+              globalStyles.boxShadow,
+              globalStyles.flexDirectionRow,
+              globalStyles.alignItemsCenter,
+              globalStyles.justifyContentCenter,
+              { backgroundColor: colors.whiteColor, borderColor: colors.line },
+            ]}
+          >
+            <ClockIcon style={globalStyles.mr2} />
+            <Text style={globalStyles.fs12}>{duration}</Text>
+          </View>
+        )}
```

Comment:
Extract this to a separate component, pls

Reply:
Extracted